### PR TITLE
JDK-8296149: Start of release updates for JDK 21

### PR DESCRIPTION
**Pull request by jddarcy**
Usual start-of-release updates. Symbol updates in initial version reflect JDK 20 build 21.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8296149](https://bugs.openjdk.org/browse/JDK-8296149): Start of release updates for JDK 21
 * [JDK-8296150](https://bugs.openjdk.org/browse/JDK-8296150): Add SourceVersion.RELEASE_21
 * [JDK-8296151](https://bugs.openjdk.org/browse/JDK-8296151): Add source 21 and target 21 to javac
 * [JDK-8296182](https://bugs.openjdk.org/browse/JDK-8296182): Start of release updates for JDK 21 (**CSR**)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [5810345e](https://git.openjdk.org/jdk/pull/10924/files/5810345e2711b980f81bb4dd9aeb590fcfab23bc)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [89731cb2](https://git.openjdk.org/jdk/pull/10924/files/89731cb2dabfdfa1a907c8155019be35f13d2f99)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [5810345e](https://git.openjdk.org/jdk/pull/10924/files/5810345e2711b980f81bb4dd9aeb590fcfab23bc)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [5810345e](https://git.openjdk.org/jdk/pull/10924/files/5810345e2711b980f81bb4dd9aeb590fcfab23bc)
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) ⚠️ Review applies to [515002c7](https://git.openjdk.org/jdk/pull/10924/files/515002c747977c1ae58a70c5450f0f617cb52896)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10924/head:pull/10924` \
`$ git checkout pull/10924`

Update a local copy of the PR: \
`$ git checkout pull/10924` \
`$ git pull https://git.openjdk.org/jdk pull/10924/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10924`

View PR using the GUI difftool: \
`$ git pr show -t 10924`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10924.diff">https://git.openjdk.org/jdk/pull/10924.diff</a>

</details>

### CODE REVIEW

**Comment by jddarcy** [comment]
Flipping the draft-bit off for this PR ahead of the upcoming start of JDK 21.

Symbol information now up-to-date as of JDK 20 b25.

File: src/java.compiler/share/classes/javax/lang/model/util/AbstractAnnotationValueVisitor14.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
  * @see AbstractAnnotationValueVisitor9
  * @since 14
  */
-@SupportedSourceVersion(RELEASE_20)
+@SupportedSourceVersion(RELEASE_21)
```
Comment (dholmes-ora) [inline src/java.compiler/share/classes/javax/lang/model/util/AbstractAnnotationValueVisitor14.java:47]: It is a pity these have to be updated each release. Is there no way to say `latest` here?

**Review by dholmes-ora — APPROVED** [review]
I looked at everything bar the sym files. All seems fine/usual.

One query below (probably one I ask each release).

Thanks.

File: src/java.compiler/share/classes/javax/lang/model/util/AbstractAnnotationValueVisitor14.java (hunk 1 above)
Comment (jddarcy) [reply src/java.compiler/share/classes/javax/lang/model/util/AbstractAnnotationValueVisitor14.java:47]: Not with the current language specification, no.

The only enum value "constant-enough" to use in an annotation is the name of an enum constant, a public static final field assigned an enum value is not enough.

On the flip side, I make the updates to these annotations using a one-line sed script I have saved away in a comment on one of the previous start-of-release-updates-for-JDK-N bugs. Using the annotations in this way is in lieu of adding another type specification for the JDK version. In other words, having AbstractAnnotationValueVisitor**14**.java with an annotation indicating RELEASE_21 prevents having AbstractAnnotationValueVisitor15, AbstractAnnotationValueVisitor16, ... , AbstractAnnotationValueVisitor21.

New visitors are now only introduced when adding new kinds of structures requires differences in behavior.

Thanks for the review.

**Review by irisclark — APPROVED** [review]

**Review by erikj79 — APPROVED** [review]

**Review by vicente-romero-oracle — APPROVED** [review]
looks good

**Review by irisclark — APPROVED** [review]

File: test/langtools/tools/javac/versions/Versions.java [hunk 2]
```diff
@@ -88,7 +88,8 @@ static enum SourceTarget {
         SEVENTEEN(false, "61.0", "17", Versions::checksrc17),
         EIGHTEEN(false,  "62.0", "18", Versions::checksrc18),
         NINETEEN(false,  "63.0", "19", Versions::checksrc19),
-        TWENTY(false,  "64.0", "20", Versions::checksrc20);
+        TWENTY(false,  "64.0", "20", Versions::checksrc20),
```
Comment (turbanoff) [inline test/langtools/tools/javac/versions/Versions.java:91]: nit:
```suggestion
        TWENTY(false, "64.0", "20", Versions::checksrc20),
```

**Review by irisclark — COMMENTED** [review]
Still looks good.

**Pull request closed without merging**